service/history/task/cross_cluster_source_task_executor.go

```diff
@@ -269,6 +281,84 @@ func (t *crossClusterSourceTaskExecutor) executeCancelExecutionTask(
 	}
 }
 
+func (t *crossClusterSourceTaskExecutor) executeApplyParentClosePolicyTask(
+	ctx context.Context,
+	task *crossClusterSourceTask,
+) (retError error) {
+	taskInfo := task.GetInfo().(*persistence.CrossClusterTaskInfo)
+	wfContext, mutableState, release, err := loadWorkflowForCrossClusterTask(ctx, t.executionCache, taskInfo, t.metricsClient, t.logger)
+	if err != nil || mutableState == nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	verified, err := task.VerifyLastWriteVersion(mutableState, taskInfo)
+	if err != nil || !verified {
+		return err
+	}
+
+	if task.ProcessingState() == processingStateInvalidated {
+		return t.generateNewTask(ctx, wfContext, mutableState, task)
+	}
+
+	// handle common errors
+	failedCause := task.response.FailedCause
+	if failedCause != nil {
+		if _, ok := expectedCloseErrors[*failedCause]; !ok {
+			return errUnexpectedErrorFromTarget
+		}
+	}
+
```

Comment:
We still need to handle `CrossClusterTaskFailedCauseDomainNotActive` here right? by generating new transfer/cross cluster tasks. Same for recordChildCompletion method below.